pkg/resourcequota/quota_validate.go

```diff
@@ -29,33 +26,35 @@ func GetProjectLock(projectID string) *sync.Mutex {
 	return mu
 }
 
-func IsQuotaFit(nsLimit *v32.ResourceQuotaLimit, nsLimits []*v32.ResourceQuotaLimit, projectLimit *v32.ResourceQuotaLimit) (bool, string, error) {
+func IsQuotaFit(nsLimit *v32.ResourceQuotaLimit, nsLimits []*v32.ResourceQuotaLimit, projectLimit *v32.ResourceQuotaLimit) (bool, api.ResourceList, error) {
 	nssResourceList := api.ResourceList{}
 	nsResourceList, err := ConvertLimitToResourceList(nsLimit)
 	if err != nil {
-		return false, "", err
+		return false, nil, err
 	}
 	nssResourceList = quota.Add(nssResourceList, nsResourceList)
 
 	for _, nsLimit := range nsLimits {
 		nsResourceList, err := ConvertLimitToResourceList(nsLimit)
 		if err != nil {
-			return false, "", err
+			return false, nil, err
 		}
 		nssResourceList = quota.Add(nssResourceList, nsResourceList)
 	}
 
 	projectResourceList, err := ConvertLimitToResourceList(projectLimit)
 	if err != nil {
-		return false, "", err
+		return false, nil, err
 	}
 
-	allowed, exceeded := quota.LessThanOrEqual(nssResourceList, projectResourceList)
-	if allowed {
-		return true, "", nil
+	_, exceeded := quota.LessThanOrEqual(nssResourceList, projectResourceList)
+	// Include resources with negative values among exceeded resources.
+	exceeded = append(exceeded, quota.IsNegative(nsResourceList)...)
```

Comment:
If the UI was working correctly, would there ever be a need for this? Why should values be negative?

Reply:
Two reasons:
1. I don't know if this gets released before the UI fix, so we could be allowing negative values, even if for a short while.
2. Users can still put negative values in the annotation before submitting a manifest directly via kubectl.